contrib/load_key_clevis_hook.sh

```diff
@@ -0,0 +1,189 @@
+#          user property latchset.clevis:jwe of the encrypted dataset for next boots
+# - Then module returns the control back to ZBM
+#
+# arg1: ZFS filesystem
+# prints: nothing
+# asks: passphrase
+# returns: 0 on success, 1 on failure
+#
+
+get_fs_value()
+{
+        fs="$1"
+        value=$2
+
+        zfs get -H -ovalue "$value" "$fs" 2> /dev/null
+}
+
+reseal_data_set()
+{
+  read -p "We have autodecrypt flag set to on, however $1 can't be unlocked. Would you like to reseal the password [yes/no] " SRESEAL
+  seq=3
+  if [[ "$SRESEAL" == "yes" ]]; then
+    while [[ "$seq" -gt 0 ]]
+    do
+      read -s -p  "Type the password, please, to unlock $1. You have $seq attempts left : " PASS
+      echo -n "$PASS" | zfs load-key -n -L prompt "$1" >&2
+      res=$?
+      if [[ "$res" == "0" ]]; then
+        seq=0
+      else
+        seq=$((seq - 1))
+        PASS=""
+      fi
+    done
+    echo "$PASS"
+    return 0
+  else
+    # In fact we don't want to reseal
+    echo ""
+    return 1
+  fi
+}
+
+load_key_clevis() {
+
+  dataset_for_clevis_unlock="$1"
+  zdebug "Processing dataset $dataset_for_clevis_unlock"
+
+  # We need to setup network for remote access
+  # We use latchset.clevis:netconf property in encrypted dataset to store net config
+
+  if [[ ! -f "/tmp/clevis_net" ]]; then # Reconfigure network only once
+    CLEVIS_NET="$(get_fs_value "${dataset_for_clevis_unlock}" "latchset.clevis:netconf")"
+    if [[ "$CLEVIS_NET" != "-" ]]; then
+      # We have net config. Reconfigure network
+      : > /tmp/clevis_net
+      IFS=':' read -r -a netconf <<< "$CLEVIS_NET"
+      dev="${netconf[0]}"
+      ip="${netconf[1]}"
+      dr="${netconf[2]}"
+      dns="${netconf[3]}"
+      zdebug "Reconfigure network: dev: $dev ip: $ip default route: $dr dns: $dns"
+      ip addr add "$ip" brd + dev "$dev"
+      ip route add default via "$dr"
+      echo "nameserver $dns" > /etc/resolv.conf
+    fi
+  fi
+
+
+  CLEVIS_CHECK="$(zfs get -H -p -o value latchset.clevis:decrypt -s local $dataset_for_clevis_unlock)"
+  if [[ "$CLEVIS_CHECK" == "yes" ]]; then
+    zdebug "Found dataset for clevis unlocking: $dataset_for_clevis_unlock"
+    KEYLOCATION="$(get_fs_value "${dataset_for_clevis_unlock}" keylocation)"
+    if [[ -z "$(echo "$KEYLOCATION"|grep file)" ]]; then
+        # That's not us
+        zdebug "Warning: keylocation is not file while clevis unlock is set for dataset $dataset_for_clevis_unlock"
+        return 0
+    fi
+    # We suppose the keylocation has value in format file:///something/key
+    KEYLOCATION="${KEYLOCATION:7}"
+    KEYSTATUS="$(zfs get -H -p -o value keystatus -s none $dataset_for_clevis_unlock)"
+    if [[ "$KEYSTATUS" == "unavailable" ]]; then
+        # Prepare the key with password for unlocking in right place
+        mkdir -p "$(dirname "$KEYLOCATION")"
+        JWE="$(zfs get -H -p -o value latchset.clevis:jwe -s local $dataset_for_clevis_unlock)"
```

Comment:
```suggestion
        JWE="$(zfs get -H -p -o value latchset.clevis:jwe -s local "$dataset_for_clevis_unlock")"
```
You should also monitor the return value of this command and take appropriate action when it fails, instead of passing garbage to `clevis decrypt`.